packages/common/src/core/types/IHubEditableContent.ts

```diff
@@ -16,28 +17,55 @@ export interface IHubEditableContent
    */
   licenseInfo: string;
   /**
+   * @deprecated use `extendedProps.serverQueryCapability` instead
    * If the item represents a feature or map service, shows whether the service has the
    * "Query" capability enabled. This is a pre-requisite for the Hub Download System to work.
    */
   serverQueryCapability?: boolean;
   /**
+   * @deprecated use `extendedProps.serverExtractCapability` instead
    * If the item represents a feature service, shows whether the service has the "Extract"
    * capability enabled. This is a pre-requisite for Create Replica Downloads to work.
    */
   serverExtractCapability?: boolean;
   /**
+   * @deprecated use `extendedProps.server.supportedExportFormats` instead
    * If the item represents a feature service with "Extract enabled", shows the formats that
    * can be extracted from the service via the "createReplica" operation.
    */
   serverExtractFormats?: string[];
   /**
+   * @deprecated use `extendedProps.additionalResources` instead
    * links to additional resources specified in the formal item metadata
    */
   additionalResources?: IHubAdditionalResource[];
   /**
    * The schedule at which the reharvest of the item will occur
    */
   schedule?: IHubSchedule;
+
+  extendedProps?: IExtendedProps;
+}
+
+type IExtendedProps = IContentExtendedProps | IServiceExtendedProps;
+
+export interface IBaseExtendedProps {
+  kind: "content" | "service";
```

Comment:
@dbouwman I know you were wary of using the word `type` as the discriminator key as that word is so overused. I went with `kind` for now, but we can switch if you have any other ideas.

Reply:
you could use `propType` or something that's contextually relevant. Kind is also fine.